pkg/apis/machineconfiguration.openshift.io/v1/types.go

```diff
@@ -339,3 +340,70 @@ type MachineConfigPoolList struct {
 
 	Items []MachineConfigPool `json:"items"`
 }
+
+// CustomResourceDefinition for KubeletConfig
+// apiVersion: apiextensions.k8s.io/v1beta1
+// kind: CustomResourceDefinition
+// metadata:
+//   # name must match the spec fields below, and be in the form: <plural>.<group>
+//   name: kubeletconfigs.machineconfiguration.openshift.io
+// spec:
+//   # group name to use for REST API: /apis/<group>/<version>
+//   group: machineconfiguration.openshift.io
+//   # list of versions supported by this CustomResourceDefinition
+//   versions:
+//     - name: v1
+//       # Each version can be enabled/disabled by Served flag.
+//       served: true
+//       # One and only one version must be marked as the storage version.
+//       storage: true
+//   # either Namespaced or Cluster
+//   scope: Cluster
+//   names:
+//     # plural name to be used in the URL: /apis/<group>/<version>/<plural>
+//     plural: kubeletconfigs
+//     # singular name to be used as an alias on the CLI and for display
+//     singular: kubeletconfig
+//     # kind is normally the CamelCased singular type. Your resource manifests use this.
+//     kind: KubeletConfig
+//     # shortNames allow shorter string to match your resource on the CLI
+//     shortNames:
+//
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KubeletConfig describes a customized Kubelet configuration.
+type KubeletConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   KubeletConfigSpec   `json:"spec,omitempty"`
+	Status KubeletConfigStatus `json:"status,omitempty"`
+}
+
+// MachineConfigPoolSelector the name of the MachineConfigPool
+type MachineConfigPoolSelector struct {
+	Name string `json:"name"`
+}
+
+// KubeletConfigSpec defines the desired state of KubeletConfig
+type KubeletConfigSpec struct {
+	MachineConfigPoolSelector MachineConfigPoolSelector                  `json:"machineConfigPoolSelector"`
```

Comment:
Sorry.... I'll get the label selector in.

Reply:
@abhinavdahiya Looks like we may need to use a FieldSelector since there not any labels on MachineConfigPools:

```
# oc get machineconfigpool  --show-labels
NAME      AGE       LABELS
master    16m       <none>
worker    16m       <none>
```